### ISSUE TITLE
Missing information on how to translate this software

### ISSUE THREAD

**Issue by scootergrisen** [ImageMagick/ImageMagick#1330]
I suggest adding information/instructions on how to translate this software.
Even if it can not be translated.
I see https://www.imagemagick.org/source/locale.xml and english.xml and francais.xml but can't seem to find any other stuff on translating.

If it it not possible to translate the software then it would be writte. "This software can't be translated at the moment" or something like that.

It could be in the README with a "Translation" section. A lot of project have a contribute page where people can read about it.

**Comment by urban-warrior** [comment]
Translation is fairly straight-forward.  Copy `english.xml` to _locale_.xml.  Change any messages from English to the target language.  As an example:

```
<message name="UnableToOpenBlob">
  unable to open image
</message>
```
becomes
```
<message name="UnableToOpenBlob">
  impossible d'ouvrir l'image
</message>
```
in French.  You then need to copy the _locale_.xml to the same location as `english.xml`, e.g. /`usr/local/ImageMagick-7/_locale_.xml`.

The final step is to contribute _locale_.xml to ImageMagick so we can include it in a future release of ImageMagick so the community can benefit from the translation.

**Comment by scootergrisen** [comment]
Maybe straight-forward but already by saying "copy english.xml" there is a problem because there are more then one english.xml and they are not identical.

Each string in the xml files starts with newline + multiple spaces and ends with newline + multiple spaces which seems incorrect.

If the software is really going to be translatable i suggest using a format like pot/po files for the translators.
That way the translators might already know the format and can use many existing tools to translate the software and check the file for errors.

When the translator later on want to update a translation the translator just needs a updated pot file and can use that to update an existing translation without having to download all the source code etc.

**Comment by urban-warrior** [comment]
Copy the english.xml file in the ImageMagick-*/config folder.  ImageMagick was written many years ago when pot/po was not supported well across platforms (Linux, Mac, Windows XP, Windows 95, etc.), so we developed our own cross-platform support.  We encourage you to contribute pot/po files along with source code support within ImageMagick to replace the existing XML-based locale files.  In addition, the pot/po files must work across all platforms, particularly Linux, Mac OS X, and Windows.  When your work is complete, post a pull request here and we will consider adding them to the ImageMagick master.

**Comment by scootergrisen** [comment]
If i was to translate the software i would start with the GUI but those strings does not seem to be in the xml file.

**Comment by urban-warrior** [comment]
Correct.  The GUI was built before we added translation messages.  The messages are exceptions.  We could easily extend it to other subsystems as well.
